### ISSUE TITLE
[WAC] - 2306 fails to load properly in Firefox

### ISSUE THREAD

**Issue by cmjdiff** [MicrosoftDocs/Windows-Admin-Center-Ideas-and-Feedback#290]
**Gateway Version**: 2306

**To Reproduce**
1. Navigate to gateway using recent Firefox (tested in 116.0.2)

**Expected behavior**
Gateway loads

**Screenshots & Additional context**
Loading screen gets stuck on "Gathering environment info—this may take a few moments."  Nothing in the browser console is of any use to me in understanding what has gone wrong.
![image](https://github.com/MicrosoftDocs/Windows-Admin-Center-Ideas-and-Feedback/assets/69131295/fb464dde-6707-4e73-b185-08121fb6fea0)
![image](https://github.com/MicrosoftDocs/Windows-Admin-Center-Ideas-and-Feedback/assets/69131295/106725e3-7698-4c19-97c7-2da6eb2d0a1f)

**Comment by RebeccaMbula** [comment]
Hi,

Mozilla is not supported and the exensions supported are Edge or Chrome.

Kind regards!

**Comment by Johan2W** [comment]
The same issue here, tested with Firefox 119.0 (64-bits). Clearing cache and cookies + starting Firefox in safe mode does not make a difference. The same Windows Admin Center loads fine in Edge.

**Comment by drkrieger01** [comment]
I too am running into this same issue with Firefox, but now also with Chrome & Edge. While there's been community provided fixes (ex.: Modifying the .js file that contains the malformed string, then re-compressing the ux.zip file in the program files directory). 

What's happening is the web page hangs at 'Gathering Environment Info..." permanently.
I do know this is likely related to the issue of Firefox running its own certificate store, but this shouldn't be a penalty against Firefox - most Systems Administrators use Firefox _because_ of this feature.

**Comment by crazyyzarc** [comment]
I also have the problem. Why is Firefox not supported?

**Comment by marcxm** [comment]
Yes, why exactly is Firefox not supported? what there is to support, aside from standard HTML and scripting? it looks like Microsoft is trying to force people to use Edge, which is based on Chrome, so they cannot do the same to Chrome.

**Comment by jefmes** [comment]
This is still true as of 4/21/2024, now using Firefox version 125.0.1. This feels extremely anti-competitive and has been going on long enough to seem intentional by Microsoft and the WAC team.

**Comment by drkrieger01** [comment]
Latest edition of WAC (2410?) also doesn't support Firefox for those who websearch this problem. 
They've gone so far as to remove the code that you can manipulate to make it work in firefox ('window.location.ancestorOrigins' renamed to 'window.location.ancestororigins' in one of the *.js files); none of the js files contain this code snipet. 

I too feel that this is anti-competitive action by MS or the WAC dev team. When administrators would like to continue using privacy focused tools, why are we then forced to use apps that are not? It forces us to create dedicated clean environment to utilize your tools; not the most effective use or investment of time.

**Comment by sockless-coding** [comment]
You can still fix it by editing `c:\ProgramData\WindowsAdminCenter\UX\349.6ffd9a23bd9692d7.js`
And changing `Array.from(window.location.ancestorOrigins)` to `Array.from(window.location.ancestorOrigins||[])`

**Comment by ross-lovell** [comment]
I'm using Firefox 136.0.1 (aarch64) on a Mac and I'm experiencing the same issue. Windows Admin Center keeps freezing at the 'Gathering environment info' stage and won't load. What's the ETA on getting this offically fixed rather than using workarounds?

**Comment by barelyprofessional** [comment]
> > It's still possible to get it working with Firefox
> > ## Fix the loading:
> > Open `c:\ProgramData\WindowsAdminCenter\UX\349.6ffd9a23bd9692d7.js` And change `Array.from(window.location.ancestorOrigins)` to `Array.from(window.location.ancestorOrigins||[])`
> > ## Fix the top navigation:
> > Open `c:\ProgramData\WindowsAdminCenter\UX\styles.e9a25a6547bf839b.css` And remove `.sme-layout-float-left{float:left}`
> 
> [@sockless-coding](https://github.com/sockless-coding) You've inspired me to make a quick workaround as the filenames for this keep changing.
> 
> [Add-FFSupportToWindowsAdminCenter](https://github.com/HerdHeaven/Admin-Scripts/blob/main/Add-FFSupportToWindowsAdminCenter.ps1)

Thank you for doing the work that the 3 Trillion Dollar company can't be bothered to. What an embarrassment of a company Microsoft is, but thankfully hiring people who can't figure out how to make their product work on more than one browser means that the line will continue to go up and that is a Good Thing.

**Comment by amirrezaalavi** [comment]
> It's still possible to get it working with Firefox
> ## Fix the loading:
> 
> Open `c:\ProgramData\WindowsAdminCenter\UX\349.6ffd9a23bd9692d7.js` And change `Array.from(window.location.ancestorOrigins)` to `Array.from(window.location.ancestorOrigins||[])`

In my version (`1.5.2407.23001`) the text is located in 

`C:\ProgramData\Server Management Experience\Ux\709.b5ce229924b023a2.js`

Also edit both files with administrative access.
